### ISSUE TITLE
Default authorization not being passed to generated actions

### ISSUE THREAD

**Issue by aaroncox** [EOSIO/eosjs#301]
I haven't been able to get the owner key working as a global option on the `Eos` object by passing it in the `authorization` parameter. At it's core, I'd like to be able to pass the authorization to the constructor itself and have that permission used for any actions issued through it. 

The following test case fails and highlights the issue:

```
  it('transfer custom authorization default (no broadcast)', () => {
    const eos = Eos({signProvider, authorization: 'inita@owner'})
    return eos.transfer('inita', 'initb', '1.0000 SYS', '')
    .then(({transaction}) => {
      const ans = [
        {actor: 'inita', permission: 'owner'}
      ]
      assert.deepEqual(transaction.transaction.actions[0].authorization, ans)
    })
  })
```

It's providing a default `authorization` of `inita@owner` upon instantiation, but when the transaction is returned, it appended the `inita@active` permission by default. From what I've seen, the options being passed in `write-api.js` on line 141 don't contain this value, but the `config` variable passed to the `WriteApi` on line 66 does. 

Is there a better way to do this or is this a legit bug?

**Comment by jcalfee** [comment]
Move authorization to here:
```js
eos.transfer('inita', 'initb', '1.0000 SYS', {authorization: 'inita@owner'})
```

**Comment by aaroncox** [comment]
I understand that appending it in the options of an action does set it properly - but it'd be ideal if it could also be a global property set upon instantiating the class.

My thought is that I'd like to be able to set the authorization at the same time I'm setting the keys, since at that point in time I have the key and can determine it's authorization.

**Comment by jcalfee** [comment]
I wonder what if anything could go wrong?  Can you think of any way a user could sign something with the wrong key or with the wrong permission level?  It would matter less if the trx got rejected.

**Comment by aaroncox** [comment]
I imagine if they ended up trying to pass the a key with an invalid authorization they're just going to get errors from the APIs back, but from what we've seen, that's happening a lot already. It's very common that we end up with users in chat who have imported their owner key into our wallet, try to perform an action, and since the authorization automatically sets to `${account_name}@active` it won't allow the user to actually perform any actions. 

To allow those actions - we'd have to go through and modify every single one of our eosjs action calls to also include that additional optional parameter. 

That scenario is a terrible example, since owner keys should really be kept offline, but the same scenario is going to occur as soon as users can create a `voting` authority which allows for BP and maybe referendum voting. They'll use that key within their everyday wallet, the wallet should recognize what permissions it grants, and then allow them access to those features. 

To address the default authorization in your second comment - why wouldn't it just keep doing that by default, but if a global was specified, it would overwrite that default authorization generation? Going further, specifying as a parameter to the action call itself could overwrite that global as well, just like it currently does for the default no-authorization configuration. 

To close this out - I'm just thinking all this through, and maybe you're onto something better I'm not just quite understanding just yet. Thanks for engaging on this 👍

**Comment by jcalfee** [comment]
Sure, we can add it.. You should probably reject it if they try to use an owner key though unless it is really needed.  The active key is better in-case it gets hacked.

**Comment by aaroncox** [comment]
Agreed on that front. We have been discouraging (and disabled) the use of the owner key for the most part - but now that we're allowing permissions management, there are a few edge cases where users will have the owner key imported.

**Comment by jcalfee** [comment]
What is your time-frame on needing this?  Do I have a week or so to release it?  I can release sooner if you need it.

This is how I would describe the feature in the [configuration](https://github.com/eosio/eosjs#configuration) section:

```markdown
* **authorization** - replace the default authorization on actions.  An
  authorization provided here may still be over-written by specifying an
  authorization for each individual action.

  For example, if most actions in an dapp are based on the posting key, this
  would replace the default active authorization with a posting authorization:

  {authorization: '@posting'}

```

**Comment by aaroncox** [comment]
No rush - but we'll upgrade as soon as it's released and do some testing. Our more advanced permissions management stuff is blocked by some missing APIs surrounding linkauths (https://github.com/EOSIO/eos/issues/5255) - so we're not only waiting for this change 👍

**Comment by jcalfee** [comment]
fixed in eosjs@16.0.3
